torch_np/random.py

```diff
@@ -0,0 +1,120 @@
+"""Wrapper to mimic (parts of) np.random API surface.
+
+NumPy has strict guarantees on reproducibility etc; here we don't give any.
+
+Q: default dtype is float64 in numpy
+
+"""
+from math import sqrt
+
+import torch
+
+from . import asarray
+from ._detail._scalar_types import default_float_type as _default_float_type
+from ._detail import _util
+
+_default_dtype = _default_float_type.torch_dtype
+
+__all__ = ["seed", "random_sample", "sample", "random", "rand", "randn", "normal"]
+
+
+def seed(seed=None):
+    if seed is not None:
+        torch.random.manual_seed()
+
+
+def random_sample(size=None):
+    if size is None:
+        values = torch.rand(())
+        return float(values)
+    else:
+        values = torch.rand(size).to(_default_dtype)
+        return asarray(values)
+
+
+def rand(*size):
+    return random_sample(size)
+
+
+sample = random_sample
+random = random_sample
+
+
+def uniform(low=0.0, high=1.0, size=None):
```

Comment:
You may want to write this function either using `torch.uniform` or doing `low + (high - low) * random(size)` (+ some proper handling of the overflow if you see fit).

Reply:
OK, refactored this to use `torch.empty(size).uniform_(...)`.
Am not going to even wonder why  `torch.rand` but  `torch.Tensor.uniform_` and no `torch.uniform`.

Reply:
large library, organically grown... jajajaj